components/base/kinematicbase/kinematics.go

```diff
@@ -126,6 +126,9 @@ func (ddk *differentialDriveKinematics) GoToInputs(ctx context.Context, desired
 		}
 
 		if !commanded {
+			if len(ddk.model.DoF()) == 2 { // 2DOF -> position only mode
```

Comment:
It's not fully clear from the code why "if we didn't command the base, and the frame is 2dof, then we should return nil here". I had to trace a bunch of function calls to understand why.

Could we add some comments explaining what's going on here?